sql-native/src/main/java/org/ballerinalang/sql/parameterprocessor/DefaultResultParameterProcessor.java

```diff
@@ -481,57 +491,64 @@ public Object convertTimeWithTimezone(java.time.OffsetTime offsetTime, int sqlTy
+                        if (zoneInfo
+                            .get(org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_HOUR) != null) {
+                            zoneMap.put(StringUtils.fromString(
+                                    org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_HOUR),
+                                    zoneInfo.get(org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_HOUR)
+                                            .longValue());
+                        } else {
+                            zoneMap.put(StringUtils.fromString(
+                                    org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_HOUR), 0);
+                        }
+                        if (zoneInfo
+                            .get(org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_MINUTE) != null) {
+                            zoneMap.put(StringUtils.fromString(
+                                    org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_MINUTE),
+                                    zoneInfo
+                                        .get(org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_MINUTE)
                                         .longValue());
+                        } else {
+                            zoneMap.put(StringUtils.fromString(
+                                    org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_MINUTE), 0);
+                        }
+                        if (zoneInfo
+                            .get(org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_SECOND) != null) {
+                            zoneMap.put(StringUtils.fromString(
+                                    org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_SECOND),
+                                    zoneInfo
+                                    .get(org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_SECOND)
+                                    .longValue());
+                        }
+                        zoneMap.freezeDirect();
+                        timeMap.put(StringUtils.fromString(
+                                org.ballerinalang.stdlib.time.util.Constants.CIVIL_RECORD_UTC_OFFSET), zoneMap);
+                        timeMap.put(StringUtils.fromString(
+                                org.ballerinalang.stdlib.time.util.Constants.CIVIL_RECORD_TIME_ABBREV),
+                                StringUtils.fromString(offsetTime.getOffset().toString()));
+                        return timeMap;
                     } else {
-                        zoneMap.put(StringUtils.fromString(
-                                org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_MINUTE), 0);
+                        throw new ApplicationError("Unsupported Ballerina type:" +
+                            type.getName() + " for SQL Time with timezone Data type.");
```

Comment:
```suggestion
                            type.getName() + " for SQL Time with timezone data type.");
```

Reply:
I will update

Reply:
Updated. Commit id is ba7d30e7b37789ba67ba680c36f81bf3a61a4588